crates/matrix-sdk-base/src/sliding_sync/mod.rs

```diff
@@ -564,7 +575,14 @@ impl BaseClient {
                 // If this event updates the current user's membership, record that in the
                 // room_info.
                 if member.state_key() == meta.user_id.as_str() {
-                    room_info.set_state(member.membership().into());
+                    let new_state: RoomState = member.membership().into();
+                    if new_state != room_info.state() {
+                        room_info.set_state(new_state);
+                        room_info_notable_updates.insert(
+                            room_info.room_id.to_owned(),
+                            RoomInfoNotableUpdateReasons::MEMBERSHIP,
+                        );
```

Comment:
Be careful that it will override an existing entry. Instead, I recommend to use the following code:

```suggestion
room_info_notable_updates
    .entry(room_id.clone()) // get the entry
    .or_default() // if it doesn't exist, create a default one
    .insert(RoomInfoNotableUpdateReasons::MEMBERSHIP); // insert `MEMBERSHIP`
```

Reply:
Should by fixed by https://github.com/matrix-org/matrix-rust-sdk/pull/3717/commits/ea949a4408d48d82a8ee34afd999825c63f7ce64